### PR TITLE
Add Workflow definition build rule (XamlAppDef)

### PR DESCRIPTION
**Pull request by wjk**
Does mostly what the title says. Some notes:

* I copied the `XamlAppDef.xaml` file from `None.xaml`, because I didn't think that any extra properties would be required. If additional properties needs to be placed in this file, please let me know.
* When I load a project that uses this functionality in the VS Experimental Instance, I still see the issue described in https://github.com/dotnet/project-system/issues/3722#issuecomment-408543998. I think this is because the updated design-time targets are not being referenced properly. Again, please let me know if I've missed anything here.

Thanks!

### CODE REVIEW

**Comment by jp2masa** [comment]
> When I load a project that uses this functionality in the VS Experimental Instance, I still see the issue described in https://github.com/dotnet/project-system/issues/3722#issuecomment-408543998.

Did you try setting the `Context` to `File;BrowseObject`?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets [hunk 1]
```diff
@@ -171,6 +171,10 @@
       <Context>File</Context>
     </PropertyPageSchema>
     
+    <PropertyPageSchema Include="$(ManagedXamlResourceDirectory)XamlAppDef.xaml">
+      <Context>File</Context>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets]: This needs to be 'File;BrowseObject'.

**Comment by davkean** [comment]
F5 will correctly setup the environment to pick up new rules + targets, so if Experimental Instance isn't working, then the bug hasn't been fixed.

**Comment by wjk** [comment]
Now this is really weird...

I added `BrowseObject` to the XAML file as @jp2masa suggested (in commit 71cb46d), and yet the Experimental Instance still doesn't show the file! I looked inside the Extensions folder for the Exp root-suffix, and none of the design-time targets files show up anywhere! Is this expected, or is something else going wrong?

**Comment by davkean** [comment]
[This](https://github.com/dotnet/project-system/blob/master/src/ProjectSystemSetup/Properties/launchSettings.json#L10) should cause us to pick it up from the recently build artifacts folder. Look in the Output window of the debugger for CPS tracing that might indicate what's going on. I can look latter to see what's going on.

**Comment by wjk** [comment]
I'll do so tomorrow, after I get off work. Thanks!

**Comment by wjk** [comment]
@davkean I've looked at the Output window, and see nothing that looks like CPS tracing. All I see are a lot of rote assembly load/unload log statements. [Here's the full log](https://github.com/dotnet/project-system/files/2246115/vs.log.txt).

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.-->
+<Rule
+  Name="XamlAppDef"
+  DisplayName="General"
+  PageTemplate="generic"
+  Description="File Properties"
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml]: This is missing PropertyPagesHidden="true"

**Comment by davkean** [comment]
I'm debugging this to see why this is occuring.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets [hunk 3]
```diff
@@ -171,6 +171,10 @@
       <Context>File</Context>
     </PropertyPageSchema>
     
+    <PropertyPageSchema Include="$(ManagedXamlResourceDirectory)XamlAppDef.xaml">
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets]: This property is missing an "s"; should be `$(ManagedXamlResourcesDirectory)`.

Comment (wjk) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets]: Fixed, but it still doesn't seem to work. Please keep looking. Thanks!

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      Description="The namespace into which the output of the custom tool is placed." />
+
+  <StringProperty
+      Name="Identity"
+      Visible="false"
+      ReadOnly="true"
+      Category="Misc"
+      Description="The item specified in the Include attribute.">
+    <StringProperty.DataSource>
+      <DataSource Persistence="Intrinsic" ItemType="None" PersistedName="Identity" SourceOfDefaultValue="AfterContext" />
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml]: Replace every usage of `ItemType="None"` with `ItemType="XamlAppDef"`.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml]: Even better, delete the `ItemType` from all properties. This is automatically pulled from the DataSource on the rule itself.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml]: Filed to https://github.com/dotnet/project-system/issues/3815 to basically do this across all the rules so we never run into this again.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml]: Also filed https://github.com/dotnet/project-system/issues/3816. We have tracing for a bunch of this, but we hook up too late for it to be output into the Output window.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml]: Strike removing the ItemType, I'm replacing it with XamlAppDef. It's required for the `BrowseObject` but not for the `File` context.

**Comment by davkean** [comment]
With the latest changes I've just pushed, this works. I'll merge it in when we are green.

**Pull request merged**